src/groups.jl

```diff
@@ -4,7 +4,15 @@
 
 immutable BenchmarkGroup
     tags::Vector{Any}
-    data::Dict{Any,Any}
+    data::Associative
+
+    function BenchmarkGroup(tags, data::Dict)
+        new(convert(Vector{Any}, tags), convert(Dict{Any, Any}, data))
+    end
+
+    function BenchmarkGroup(tags, data)
+        new(convert(Vector{Any}, tags), data)
+    end
 end
```

Comment:
> there are some tests that construct other dicts (e.g. Dict{String, Trial} and Dict{String, TrialEstimate}) 

If tests are failing just due to over-specialized `BenchmarkGroup` initialization, they should be using the `BenchmarkGroup(::Vector, ::Pair...)` constructor (which should be changed per my comment below).

> and some code that assume .data is {Any, Any}.

That code doesn't assume that `data::Dict{Any, Any}`, just that `typeof(k) <: keytype(dest.data)` and `typeof(f($(getinds...))) <: valtype(dest.data)` (e.g. like `map!` for arrays). There might be other code that actually does assume `Dict{Any,Any}`, though, in which case this PR would have to generalize that code.

This PR is all about allowing the user to select the their own `Associative` type - we shouldn't special case `Dict`s to break that behavior.